src/omv/imlib/jpege.c

```diff
@@ -1645,10 +1645,16 @@ bool jpeg_compress(image_t *src, image_t *dst, int quality, bool realloc) {
     jpeg_subsample_t jpeg_subsample = JPEG_SUBSAMPLE_1x1;
 
     if (src->is_color) {
-        if (quality <= 35) {
-            jpeg_subsample = JPEG_SUBSAMPLE_2x2;
-        } else if (quality < 60) {
+        if (subsampling == JPEG_SUBSAMPLING_AUTO) {
+            if (quality <= 35) {
+                jpeg_subsample = JPEG_SUBSAMPLE_2x2;
+            } else if (quality < 60) {
+                jpeg_subsample = JPEG_SUBSAMPLE_2x1;
+            }
+        } else if (subsampling == JPEG_SUBSAMPLING_422) {
             jpeg_subsample = JPEG_SUBSAMPLE_2x1;
+        } else if (subsampling == JPEG_SUBSAMPLING_420) {
+            jpeg_subsample = JPEG_SUBSAMPLE_2x2;
```

Comment:
```suggestion
        if (subsampling == JPEG_SUBSAMPLING_AUTO) {
            if (quality <= 35) {
                subsampling = JPEG_SUBSAMPLING_420;
            } else if (quality < 60) {
                subsampling = JPEG_SUBSAMPLING_422;
            }
        }
```

We won't need the old constants anymore, or the `jpeg_subsample` variable.

Reply:
Done